File: skills/astra-toolkit/clients/csharp/examples/admin-find-embedding-providers.cs
using DataStax.AstraDB.DataApi;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing database
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      System.Environment.GetEnvironmentVariable("ASTRA_DB_API_ENDPOINT"),
      System.Environment.GetEnvironmentVariable("ASTRA_DB_APPLICATION_TOKEN")
    );

    var databaseAdmin = database.GetAdmin();

    var result = await databaseAdmin.FindEmbeddingProvidersAsync();

    var providers = result.EmbeddingProviders;
  }
}

File: skills/astra-toolkit/clients/python/examples/admin-find-embedding-providers.py
import os
from astrapy import DataAPIClient

client = DataAPIClient(os.environ["ASTRA_DB_APPLICATION_TOKEN"])

admin = client.get_admin()

database_admin = admin.get_database_admin(os.environ["ASTRA_DB_API_ENDPOINT"])

providers = database_admin.find_embedding_providers()

# Use the raw dict
print(providers.raw_info)

# Or work with the resulting object:

print(providers.embedding_providers.keys())

print(providers.embedding_providers["openai"])

print(providers.embedding_providers["openai"].parameters)

print(providers.embedding_providers["openai"].supported_authentication)

print(providers.embedding_providers["openai"].models[0])

print(providers.embedding_providers["openai"].models[0].parameters)
